### ISSUE TITLE
Child WorkflowJob not started when Inventory on WorkflowJobTemplateNode

### ISSUE THREAD

**Issue by slemrmartin** [ansible/awx#5046]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
I have nested workflow templates, when I set Prompt on launch on child workflow template and set Inventory through "Prompt" in Workflow visualizer, then Launch is in Running state forever and Workflow job for child Workflow template doesn't exist (is probably deleted, because there is missing ID in list of Jobs)

##### ENVIRONMENT
* AWX version: Tower 3.5.0
* AWX install method: AMI -  https://www.ansible.com/products/tower/trial - Tower in Amazon EC2
* Ansible version:  2.8.0
* Operating System: 
* Web Browser: Google Chrome 77.0.3865.120

##### STEPS TO REPRODUCE

Child Workflow template contains: 
- No inventory, Prompt on launch
- Job Template with https://github.com/mkanoor/playbook/blob/master/hello_world.yml
  - No inventory, Prompt on launch
- Workflow Job Template Node with Inventory1 (added through "Prompt" button in Workflow visualizer)

Parent Workflow template contains:
- No inventory
- Child workflow
  - (No inventory, Prompt on Launch)
- Workflow Job Template Node with Inventory2  (added through "Prompt" button in Workflow visualizer)

```
Parent Workflow (without Inventory)
(Start) ==> (Inventory2 on Workflow Node) ==> [ 
      Child Workflow (without Inventory, Prompt on Launch) =>
          (Start) ==> (Inventory1 on Workflow Node) ==> Job Template with Prompt on Launch  
]
```
##### EXPECTED RESULTS

Job Template should be successfully finished (I suppose with Inventory2?)

##### ACTUAL RESULTS

Parent workflow job Status = Running forever
Child workflow job doesn't exist (probably deleted, one ID in Jobs is missing)
API:
(the only node) in `api/workflow_jobs/<parent_workflow_id>/workflow_nodes` doesn't have `/summary_fields/job`, also `/related/job` doesn't exist. But there are `/summary_fields/inventory` and `/related/inventory`. Parent workflow doesn't have inventory

In another case that Parent workflow has inventory and workflow node doesn't have, then `/summary_fields/job` and `/related/job` links exist and point to child workflow.

**Comment by shanemcd** [comment]
Ping @AlanCoding

**Comment by AlanCoding** [comment]
I'm only testing with AWX devel branch for now, but I'm not able to reproduce this. I launch the outer workflow, and it runs all the way down to the JT, which winds up running with inventory2, because the workflow-level inventory will be applied at higher precedence than the node-level inventory.

### parent

![Screen Shot 2019-10-21 at 3 56 28 PM](https://user-images.githubusercontent.com/1385596/67238580-68bcaa80-f41b-11e9-9fb7-2d4a8c81fc4d.png)

### child

![Screen Shot 2019-10-21 at 3 56 34 PM](https://user-images.githubusercontent.com/1385596/67238603-74a86c80-f41b-11e9-99da-fccfc033c1a0.png)

### job template

![Screen Shot 2019-10-21 at 3 56 43 PM](https://user-images.githubusercontent.com/1385596/67238620-7c681100-f41b-11e9-82d7-bcf1ccebdabf.png)

**Comment by AlanCoding** [comment]
The most useful thing we could probably get is the server logs from your particular case.

**Comment by AlanCoding** [comment]
Got more details, and replicated.

PATCH `/api/v2/workflow_job_template_nodes/5/` where the workflow the node uses prompts for variables (probably applies equally to JT nodes).

```json
{
    "extra_data": "---\n"
}
```

Launch this, and 💥 

> Child WorkflowJob not started

It's not just this. That task manager is not completing, and therefore, not processing jobs.

```
awx_1        | 2019-10-22 19:51:01,353 ERROR    awx.main.dispatch Worker failed to run task awx.main.scheduler.tasks.run_task_manager(*[], **{}
awx_1        | Traceback (most recent call last):
awx_1        |   File "/awx_devel/awx/main/dispatch/worker/task.py", line 85, in perform_work
awx_1        |     result = self.run_callable(body)
awx_1        |   File "/awx_devel/awx/main/dispatch/worker/task.py", line 61, in run_callable
awx_1        |     return _call(*args, **kwargs)
awx_1        |   File "/awx_devel/awx/main/scheduler/tasks.py", line 15, in run_task_manager
awx_1        |     TaskManager().schedule()
awx_1        |   File "/awx_devel/awx/main/scheduler/task_manager.py", line 638, in schedule
awx_1        |     self._schedule()
awx_1        |   File "/awx_devel/awx/main/scheduler/task_manager.py", line 622, in _schedule
awx_1        |     self.spawn_workflow_graph_jobs(running_workflow_tasks)
awx_1        |   File "/awx_devel/awx/main/scheduler/task_manager.py", line 118, in spawn_workflow_graph_jobs
awx_1        |     kv = spawn_node.get_job_kwargs()
awx_1        |   File "/awx_devel/awx/main/models/workflow.py", line 241, in get_job_kwargs
awx_1        |     prompts_data['extra_vars'].update(self.workflow_job.extra_vars_dict)
awx_1        | AttributeError: 'str' object has no attribute 'update'
```

**Comment by kdelee** [comment]
Followed reproduction steps:

1) create child workflow that accepts prompt on launch inventory
2) create parent workflow and put child workflow as node in workflow
3) set inventory on workflow node
4) set `extra_data` to `"---\n"` on workflow node
5) launch

The workflow runs and succeeds as expected.
Also tried including an actual job template in the child workflow -- the inventory is used and I got expected outcome of job template running against that inventory.

Closing, thanks!
